glass/fields.py

```diff
@@ -26,7 +26,7 @@
     Cls = Sequence[NDArray[Any]]
 
 
-def inv_triangle_number(triangle_number: int) -> int:
+def nfields_from_nspectra(triangle_number: int) -> int:
     r"""
     The :math:`n`-th triangle number is :math:`T_n = n \, (n+1)/2`.  If
     the argument is :math:`T_n`, then :math:`n` is returned.  Otherwise,
```

Comment:
I would update the argument and docstring here to reflect the new function name. Something along the lines of

> Returns the number of fields for a number of spectra.
>
> Given the number of spectra *nspectra*, returns the number of fields *n* such that ``n * (n + 1) // 2 == nspectra``.

In addition, I would also update the error message:

> invalid number of spectra: {nspectra}

That means there's no need to catch and rethrow the error elsewhere in fields.py

Reply:
Updated!